tests/Doctrine/Tests/ORM/Mapping/XmlMappingDriverTest.php

```diff
@@ -149,27 +149,69 @@ public function testInvalidMappingFileException()
      * @param string $xmlMappingFile
      * @dataProvider dataValidSchema
      * @group DDC-2429
+     * @group 6389
      */
     public function testValidateXmlSchema($xmlMappingFile)
+    {
+        $this->assertTrue($this->doValidateXmlSchema($xmlMappingFile));
+    }
+
+    /**
+     * @param string $xmlMappingFile
+     * @dataProvider dataInvalidSchema
+     * @group 6389
+     */
+    public function testInvalidateXmlSchema($xmlMappingFile)
+    {
+        // Avoid PHP Warning errors for expected invalid mappings
+        $savedUseErrors = libxml_use_internal_errors(true);
+
+        try {
+            $this->assertFalse($this->doValidateXmlSchema($xmlMappingFile), 'Invalid XML mapping should not pass XSD validation.');
+        } finally {
+            // Restore previous setting
+            libxml_clear_errors();
+            libxml_use_internal_errors($savedUseErrors);
+        }
+    }
+
+    private function doValidateXmlSchema($xmlMappingFile)
     {
         $xsdSchemaFile  = __DIR__ . '/../../../../../doctrine-mapping.xsd';
-        $dom            = new \DOMDocument('UTF-8');
+        $dom            = new \DOMDocument('1.0', 'UTF-8');
 
         $dom->load($xmlMappingFile);
 
-        $this->assertTrue($dom->schemaValidate($xsdSchemaFile));
+        return $dom->schemaValidate($xsdSchemaFile);
     }
 
     static public function dataValidSchema()
     {
-        $list    = glob(__DIR__ . '/xml/*.xml');
+        return self::doDataValidSchema(true);
+    }
+
+    static public function dataInvalidSchema()
```

Comment:
This sort of API is quite confusing: what's going on? Why is `doDataValidSchema()` used also for invalid data?

Reply:
Would it be enough to rename it, or do you prefer that I factorize the common parts differently (like the test functions)?

BTW, would it be a BC-break to rename an existing public test function?
E.g. `testValidateXmlSchema` to, say, `testValidateXmlSchemaWithValidMapping` (so that `testInvalidateXmlSchema` could be `testValidateXmlSchemaWithInvalidMapping`)? (maybe you have better ideas?)

Reply:
Having completely separate methods, even if some code is duplicated, is preferable here